samples/csharp/MicrosoftEduGraphSamples/Workflows/ModuleWorkflow.cs

```diff
@@ -0,0 +1,150 @@
+                }
+                else if (string.IsNullOrEmpty(_config["password"]))
+                {
+                    throw new Exception("Missing password please check appconfig.json file.");
+                }
+            }
+            catch (Exception ex)
+            {
+                throw;
+            }
+        }
+        /// <summary>
+        /// Workflow to create and publish the module
+        /// </summary>
+        public async Task ClassworkAsync()
+        {
+            try
+            {
+                // Get a Graph client using delegated permissions
+                var graphClient = GraphClient.GetDelegateClient(_config["tenantId"], _config["appId"], _config["teacherAccount"], _config["password"]);
+
+                //1 - create a draft module.
+                var module = await MicrosoftGraphSDK.Module.CreateAsync(graphClient, _config["classId"], "Module Workflow", "This is the first module created using Graph SDK.");
+                Console.WriteLine($"New module has been created: {module.Id} - {module.DisplayName} - {module.Status}");
+                await MicrosoftGraphSDK.Module.SetupResourcesFolder(graphClient, _config["classId"], module.Id);
+
+                //2 - add a link resource.
+                EducationModuleResource requestBody = new EducationModuleResource
+                {
+                    Resource = new EducationLinkResource
+                    {
+                        OdataType = "#microsoft.graph.educationLinkResource",
+                        DisplayName = "Bing site",
+                        Link = "https://www.bing.com",
+                    },
+                };
+                // A new file is added
+                var newResource = await MicrosoftGraphSDK.Module.PostResourceAsync(graphClient, _config["classId"], module.Id.ToString(), requestBody);
+
+                //3 - add a new word document resource.
+                requestBody = new EducationModuleResource
+                {
+                    Resource = new EducationWordResource
+                    {
+                        OdataType = "#microsoft.graph.educationWordResource",
+                        DisplayName = "test_word_file.docx",
+
+                    },
+                };
+                // A new Word file is added
+                newResource = await MicrosoftGraphSDK.Module.PostResourceAsync(graphClient, _config["classId"], module.Id.ToString(), requestBody);
+
+                //5 - add a channel resource
+                requestBody = new EducationModuleResource
+                {
+                    Resource = new EducationChannelResource
+                    {
+                        OdataType = "#microsoft.graph.educationChannelResource",
+                        Url = "https://graph.microsoft.com/v1.0/teams/" + _config["classId"] + "/channels/19:Z2JOIPdvU-i1gymVObk9KhJj_sZCvDFEyGXPO7jLLUA1@thread.tacv2",
```

Comment:
I think you must avoid hardcoding the channelId,
it's preferably to have an additional call to SDK to get the id and then use it here

Reply:
need help on this

Reply:
Resolved updated channel code